edb/schema/objtypes.py

```diff
@@ -100,10 +100,15 @@ def is_object_type(self) -> bool:
         return True
 
     def is_free_object_type(self, schema: s_schema.Schema) -> bool:
-        return self.issubclass(
-            schema,
-            schema.get('std::FreeObject', type=ObjectType),
-        )
+        if self.get_name(schema) == sn.QualName('std', 'FreeObject'):
+            return True
+
+        FreeObj = schema.get('std::FreeObject', type=ObjectType, default=None)
```

Comment:
nit: I'd keep the full name ;)

```suggestion
        FreeObject = schema.get('std::FreeObject', type=ObjectType, default=None)
```